examples/agentInteractionsDemo/app.js

```diff
@@ -0,0 +1,283 @@
+        let pcEnvironment = getEmbeddingPCEnv();
+        if (!pcEnvironment) {
+            this.errorMessage = 'Cannot identify App Embeddding context.  Did you forget to add pcEnvironment={{pcEnvironment}} to your app\'s query string?';
+            return;
+        }
+
+        /*
+            * Note: To use this app in your own org, you will need to create your own OAuth2 Client(s)
+            * in your PureCloud org.  After creating the Implicit grant client, map the client id(s) to
+            * the specified region key(s) in the object below, deploy the page, and configure an app to point to that URL.
+            */
+        let pcOAuthClientIds = {'mypurecloud.com': 'implicit-oauth-client-id-here'};
+
+        let clientId = pcOAuthClientIds[pcEnvironment];
+        if (!clientId) {
+            this.errorMessage = pcEnvironment + ': Unknown/Unsupported PureCloud Environment';
+            return;
+        }
+
+        let client = platformClient.ApiClient.instance;
+        client.setEnvironment(pcEnvironment);
+        client.setPersistSettings(true);
+
+        let clientApp = null;
+        try {
+            clientApp = new window.purecloud.apps.ClientApp({
+                pcEnvironment,
+            });
+            Vue.prototype.$clientApp = clientApp;
+        } catch (e) {
+            console.log(e);
+            this.errorMessage = pcEnvironment + ": Unknown/Unsupported PureCloud Embed Context";
+            return;
+        }
+
+        // Create API instance
+        const usersApi = new platformClient.UsersApi();
+        const qualityApi = new platformClient.QualityApi();
+        const conversationsApi = new platformClient.ConversationsApi();
+        Vue.prototype.$usersApi = usersApi;
+        Vue.prototype.$qualityApi = qualityApi;
+        Vue.prototype.$conversationsApi = conversationsApi;
+
+        let authenticated = false;
+        let agentUserId = "";
+
+        let redirectUrl = window.location.origin;
+        if (!redirectUrl) {
+            redirectUrl = window.location.protocol + '//' + window.location.host;
+        }
+        redirectUrl += window.location.pathname;
+
+        // Authentication and main flow
+        client.loginImplicitGrant(clientId, redirectUrl, { state: ("pcEnvironment=" + pcEnvironment) })
+            .then(() => {
+                // Get userme info
+                authenticated = true;
+                return usersApi.getUsersMe({ "expand": ["presence"] });
+            })
+            .then((profileData) => {
+                // Process agent's profile data
+                this.profileData = profileData;
+                agentUserId = profileData.id;
+                this.authenticated = true;
+
+                // Get evaluations data
+                const startTime = moment('1997-01-01').toISOString();
+                const endTime = moment().toISOString();
+                const evaluationListPromise = getEvaluations(qualityApi, startTime, endTime, agentUserId);
```

Comment:
You could remove these two promise variable references since you're not using them.  With it like it is, it seems like you forgot to return the promise results (maybe in a Promise.all).

You're ok with not waiting on the results here, right?  i.e. you want to immediately return from this and let the results trickle in and update the model, right?  In a bigger app, this might thrash the render and component calculation; but, I think it's fine here.